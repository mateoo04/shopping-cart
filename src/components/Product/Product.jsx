import { useState } from 'react';
import PropTypes from 'prop-types';

export default function Product({ imageUrl, name, price }) {
  const [quantity, setQuantity] = useState(1);

  return (
    <div>
      <img src={imageUrl} alt={name} />
      <h3>{name}</h3>
      <p>{`${price}â‚¬`}</p>
      <div>
        <button onClick={() => quantity > 0 && setQuantity(quantity - 1)}>
          -
        </button>
        <input type='number' value={quantity} readOnly />
        <button onClick={() => quantity < 99 && setQuantity(quantity + 1)}>
          +
        </button>
      </div>
      <button>ADD TO BASKET</button>
    </div>
  );
}

Product.propTypes = {
  imageUrl: PropTypes.string,
  name: PropTypes.string,
  price: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};
