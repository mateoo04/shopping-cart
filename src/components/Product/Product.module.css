.product {
  display: flex;
  flex-direction: column;
  gap: 4px;
}

.quantityPicker {
  background-color: #ebebeb;
  border-radius: 50px;
  align-self: center;
  overflow: hidden;
}

.product button {
  background-color: #000;
  color: #fff;
}

.quantityPicker input {
  border: 0;
  width: 34px;
  background: transparent;
  text-align: center;
}

.addToBasketButton {
  margin-top: 5px;
  border-radius: 50px;
  padding: 8px 0;
}
