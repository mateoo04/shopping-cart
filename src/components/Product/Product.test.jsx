import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import Product from './Product.jsx';

describe('Product component', () => {
  it('renders product correctly', () => {
    const { container } = render(
      <Product
        imageUrl={
          'https://storage.googleapis.com/fir-auth-1c3bc.appspot.com/1692947383286-714WUJlhbLS._SL1500_.jpg'
        }
        name='Sony WH-10000'
        price={189}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('renders product name correctly', () => {
    render(
      <Product
        imageUrl={
          'https://storage.googleapis.com/fir-auth-1c3bc.appspot.com/1692947383286-714WUJlhbLS._SL1500_.jpg'
        }
        name='Sony WH-10000'
        price={189}
      />
    );

    expect(screen.getByRole('heading', { level: 3 }).textContent).toMatch(
      'Sony WH-10000'
    );
  });
});
