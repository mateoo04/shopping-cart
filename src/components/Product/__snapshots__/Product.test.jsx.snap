// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Product component > renders product correctly 1`] = `
<div>
  <div>
    <img
      alt="Sony WH-10000"
      src="https://storage.googleapis.com/fir-auth-1c3bc.appspot.com/1692947383286-714WUJlhbLS._SL1500_.jpg"
    />
    <h3>
      Sony WH-10000
    </h3>
    <p>
      189â‚¬
    </p>
    <div>
      <button>
        -
      </button>
      <input
        readonly=""
        type="number"
        value="1"
      />
      <button>
        +
      </button>
    </div>
    <button>
      ADD TO BASKET
    </button>
  </div>
</div>
`;
